packages/branding/src/browser/branding-service.ts

```diff
@@ -0,0 +1,104 @@
+/********************************************************************************
+ * Copyright (C) 2019 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { BrandingStyle } from './branding-style';
+import { BrandingProps } from './branding-props';
+import { BuiltinThemeProvider } from '@theia/core/lib/browser/theming';
+
+export class BrandingService {
+
+    protected style: BrandingStyle;
+
+    protected props: BrandingProps;
+
+    protected readonly MAIN_SELECTOR = '#theia-main-content-panel';
+    protected readonly MENU_SELECTOR = '.theia-icon';
+
+    /**
+     * The default `Theia` branding properties.
+     */
+    protected static readonly DEFAULT_BRANDING_PROPS: BrandingProps = {
+        darkMain: require('../../src/browser/icons/theia-brand-logo-dark.png'),
+        darkMenu: require('../../src/browser/icons/theia-menu-logo-dark.png'),
+        lightMain: require('../../src/browser/icons/theia-brand-logo-light.png'),
+        lightMenu: require('../../src/browser/icons/theia-menu-logo-light.png'),
+        mainSize: '40%'
+    };
+
+    protected constructor() {
+        this.style = new BrandingStyle();
+        this.props = BrandingService.DEFAULT_BRANDING_PROPS;
+        this.updateBranding();
+    }
+
+    static get(): BrandingService {
+        return new BrandingService();
```

Comment:
Nope, everytime get will be called, it will create a new instance.

Reply:
thanks! it's been addressed